Enable CSP FreeRTOS build for cmake

dimitrovand opened the pull request.

yashi commented on CMakeLists.txt:
```diff
@@ -17,6 +20,8 @@ target_include_directories(freertos_config SYSTEM
 
 add_subdirectory(freertos)
+
+set(CSP_KERNEL_NAME freertos)
```
Move this to right under `set(CMAKE_SYSTEM_NAME Generic)`.

yashi reviewed (changes requested):
@dimitrovand Thanks for the PR. Please write a commit log, please.

dimitrovand committed:
Enable CSP FreeRTOS build for cmake

Set CMAKE_SYSTEM_NAME to Generic and CSP_SYSTEM_NAME to freertos
as required by libcsp.
